File: 0226-invert-binary-tree/0226-invert-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void solution(TreeNode* root){
        if(!root) return;
        solution(root->left);
         solution(root->right);
       swap(root->left,root->right);
    }
    TreeNode* invertTree(TreeNode* root) {
        solution(root);
        return root;
    }
};

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None:
            return root

        old_left = root.left

        root.left = root.right 
        root.right = old_left

        if root.left is not None:
            root.left = self.invertTree(root.left)
        
        if root.right is not None:
            root.right = self.invertTree(root.right)

        return root
